### PR TITLE
chore: align with upstream exception handling

### PR DESCRIPTION
**Pull request by mxschmitt**
https://github.com/microsoft/playwright-dotnet/issues/2969

### CODE REVIEW

File: src/Playwright/Transport/Converters/EvaluateArgumentValueConverter.cs [hunk 1]
```diff
@@ -123,6 +123,19 @@ internal static object Serialize(object value, List<EvaluateArgumentGuidElement>
             return new { bi = bigInteger.ToString(CultureInfo.InvariantCulture) };
         }
 
+        if (value is Exception exception)
+        {
+            return new Dictionary<string, object>
+            {
+                ["e"] = new Dictionary<string, object>
```
Comment (yury-s) [inline src/Playwright/Transport/Converters/EvaluateArgumentValueConverter.cs:130]: What's the point in serializing .net error to a js error? I'd not add this at all, especially given that we not parse the error from js and just return a string.

File: src/Playwright.Tests/PageEvaluateTests.cs [hunk 2]
```diff
@@ -547,24 +547,62 @@ public async Task ShouldNotLeakHandles()
...
     {
         var exception = await Page.EvaluateAsync<PlaywrightException>(@"() => {
             return (function functionOnStack() {
                 return new Error('error message');
             })();
         }");
-        Assert.Equals("error message", exception.Message);
-        StringAssert.Contains("functionOnStack", exception.StackTrace);
+        StringAssert.Contains("Error: error message", exception.Message);
+        StringAssert.Contains("functionOnStack", exception.Message);
```
Comment (yury-s) [inline src/Playwright.Tests/PageEvaluateTests.cs:558]: Shouldn't it be in exception.StackTrace ?

File: src/Playwright.Tests/PageEvaluateTests.cs [hunk 3]
```diff
@@ -547,24 +547,62 @@ public async Task ShouldNotLeakHandles()
...
-        StringAssert.Contains("Error: error message", exception);
+        var exception = await Page.EvaluateAsync<object>(@"() => {
+            function innerFunction() {
+                const e = new Error('error message');
+                e.name = 'foobar';
+                return e;
+            }
+            return innerFunction();
+        }");
+        Assert.IsInstanceOf<PlaywrightException>(exception);
```
Comment (yury-s) [inline src/Playwright.Tests/PageEvaluateTests.cs:572]: I'd use a generic Exception instead of PlaywrightException

**Review by yury-s — APPROVED** [review]

File: src/Playwright.Tests/PageEvaluateTests.cs (hunk 2 above)
Comment (mxschmitt) [reply src/Playwright.Tests/PageEvaluateTests.cs:558]: We don't re-serialise the JS error into a .NET StackTrace property yet, I can try doing it tho, we only do message as of today.

**Pull request merged**